modules/os2forms_digital_post/src/Form/SettingsForm.php

```diff
@@ -258,8 +258,27 @@ public function submitForm(array &$form, FormStateInterface $formState): void {
   private function testCertificate(): void {
     try {
       $certificateLocator = $this->certificateLocatorHelper->getCertificateLocator();
-      $certificateLocator->getCertificates();
-      $this->messenger()->addStatus($this->t('Certificate succesfully tested'));
+      $certificatePath = $this->settings->getCertificate()[CertificateLocatorHelper::LOCATOR_TYPE_FILE_SYSTEM]['path'];
+
+      // Check if the certificate has the pkcs12 extension or not.
+      if (pathinfo($certificatePath, PATHINFO_EXTENSION) == 'pkcs12') {
+        // Check the certificate if it is a valid pkcs12 certificate.
+        $certificateLocator->getCertificates();
+      }
+      else {
+        // Get contents of certificate.
+        $certificateKeyFile = $certificateLocator->getCertificate();
+        // Create an array for checking the key with the certificate.
+        $keyCheckData = [$certificateKeyFile, $certificateLocator->getPassphrase()];
+        // Check the private key against the certificate.
+        $result = openssl_x509_check_private_key($certificateKeyFile, $keyCheckData);
+        // If the result is not "1", throw an exception.
+        if ($result != 1) {
```

Comment:
returned result `$result` is of type boolean. It would be more correct to compare it against `boolean` value, instead of `int`.
If fact syntax can be simplified to:
```
if ($result) {
...
```